src/main.rs

```diff
@@ -71,34 +70,31 @@ impl Deck {
         };
     }
 
-    fn shuffle(&self) -> Deck {
-        let mut cards = self.cards.to_vec();
+    fn shuffle(mut self) -> Deck {
```

Comment:
I think this is quite cool. 
If you take a `mut self`, you can mutate and return the original instance, safe in the knowledge that variables bound to the original instance can no longer be used.
`self` is *moved* into this method, mutated and then returned; to be bound to another variable by the caller.